src/OrleansPSUtils/StartGrainClient.cs

```diff
@@ -0,0 +1,92 @@
+        public ClientConfiguration Config { get; set; }
+
+        [Parameter(Position = 4, Mandatory = true, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public IPEndPoint GatewayAddress { get; set; }
+
+        [Parameter(Position = 5, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public bool OverrideConfig { get; set; } = true;
+
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = FilePathSet)]
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = FileSet)]
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = ConfigSet)]
+        [Parameter(Position = 6, ValueFromPipeline = true, ParameterSetName = EndpointSet)]
+        public TimeSpan Timeout { get; set; } = TimeSpan.Zero;
+
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                WriteVerbose($"[{DateTime.UtcNow}] Initializing Orleans Grain Client");
+
+                switch (ParameterSetName)
+                {
+                    case FilePathSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using config file at '{ConfigFilePath}'...");
+                        if (string.IsNullOrWhiteSpace(ConfigFilePath))
+                            throw new ArgumentNullException(nameof(ConfigFilePath));
+                        GrainClient.Initialize(ConfigFilePath);
+                        break;
+                    case FileSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using provided config file...");
+                        if (ConfigFile == null)
+                            throw new ArgumentNullException(nameof(ConfigFile));
+                        GrainClient.Initialize(ConfigFile);
+                        break;
+                    case ConfigSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using provided 'ClientConfiguration' object...");
+                        if (Config == null)
+                            throw new ArgumentNullException(nameof(Config));
+                        GrainClient.Initialize(Config);
+                        break;
+                    case EndpointSet:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using default Orleans Grain Client initializer");
+                        if (GatewayAddress == null)
+                            throw new ArgumentNullException(nameof(GatewayAddress));
+                        GrainClient.Initialize(GatewayAddress, OverrideConfig);
+                        break;
+                    default:
+                        WriteVerbose($"[{DateTime.UtcNow}] Using default Orleans Grain Client initializer");
+                        GrainClient.Initialize();
+                        break;
+                }
+
+                if (Timeout != TimeSpan.Zero)
+                    GrainClient.SetResponseTimeout(Timeout);
+            }
```

Comment:
since starting the client consumes resources on the silos, should we have some idle time where we disconnect the client? Not sure what's the best approach, but it's a little bit scary to leave tons of grain clients started unknowingly even after you are no longer running any scripts. Or maybe this already happens automatically by the grain client who only keeps connections open when it needs them? /cc @jason-bragg